From b2c88f00503a748cea885d03b03c0ddb05a8c311 Mon Sep 17 00:00:00 2001
From: Thomas Stuart <tjstuart@google.com>
Date: Fri, 28 Jun 2024 17:33:03 +0000
Subject: [PATCH] enforce calling package for disabling visual voicemail

Flag: EXEMPT security patch
Bug: 382495957 (partner)
Bug: 308932906 (root)
Test: CTS
(cherry picked from commit 9aee7c7bea9d9454f98ce0e43e5ae8d1974cf483)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:45ad363af1b0e790cf513b6153b9230f0264146e)
Merged-In: I631c8f7a3dd507e7127a2060bded78cf4ec89c0b
Change-Id: I631c8f7a3dd507e7127a2060bded78cf4ec89c0b
---
 src/com/android/phone/PhoneInterfaceManager.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/com/android/phone/PhoneInterfaceManager.java b/src/com/android/phone/PhoneInterfaceManager.java
index ec15715e4..a8790ec68 100644
--- a/src/com/android/phone/PhoneInterfaceManager.java
+++ b/src/com/android/phone/PhoneInterfaceManager.java
@@ -4093,6 +4093,7 @@ public class PhoneInterfaceManager extends ITelephony.Stub {
     @Override
     public void disableVisualVoicemailSmsFilter(String callingPackage, int subId) {
         mAppOps.checkPackage(Binder.getCallingUid(), callingPackage);
+        enforceVisualVoicemailPackage(callingPackage, subId);
 
         final long identity = Binder.clearCallingIdentity();
         try {
-- 
2.48.1.262.g85cc9f2d1e-goog

